Userland/Libraries/LibWeb/CSS/StyleProperties.cpp

```diff
@@ -229,6 +229,18 @@ Optional<int> StyleProperties::z_index() const
     return static_cast<int>(value.value()->to_length().raw_value());
 }
 
+Optional<float> StyleProperties::opacity() const
+{
+    auto value = property(CSS::PropertyID::Opacity);
+    if (!value.has_value())
+        return {};
+
+    if (value.value()->to_length().is_percentage())
```

Comment:
This probably recalculates the length, how about this:
```suggestion
    if (auto length = value.value()->to_length(); length.is_percentage())
        return ...(length.raw_value())...;
    else
         return ...;
```

Reply:
`to_length` doesn't calculate anything, it just returns a value

Reply:
It's a virtual call to a pointer in an optional, that's not _free_.
And even then, it's much nicer to see `length` instead of `value.value()->to_length()` .

Reply:
oh, you didn't mean the actual length, sorry :^)

Reply:
Fixed